diff --git a/kernel/Kbuild b/kernel/Kbuild
index abd1380..6274d22 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -64,6 +64,11 @@ $(info -- $(REPO_NAME) version name: $(KSU_VERSION_FULL))
 
 ccflags-y += -DKSU_VERSION=$(KSU_VERSION)
 ccflags-y += -DKSU_VERSION_FULL=\"$(KSU_VERSION_FULL)\"
+
+ifndef RESUKISU_MANAGER_LIST
+RESUKISU_MANAGER_LIST := 0x377:d3469712b6214462764a1d8d3e5cbe1d6819a0b629791b9f4101867821f1df64,0x35c:947ae944f3de4ed4c21a7e4f7953ecf351bfa2b36239da37a34111ad29993eef,0x375:484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588,0x384:a9462b8b98ea1ca7901b0cbdcebfaa35f0aa95e51b01d66e6b6d2c81b97746d8,0x396:f415f4ed9435427e1fdf7f1fccd4dbc07b3d6b8751e4dbcec6f19671f427870b
+endif
+
+ccflags-y += -DKSU_NEXT_MANAGER_LIST=\"$(RESUKISU_MANAGER_LIST)\"
 
 ifeq ($(CONFIG_KSU_MULTI_MANAGER_SUPPORT), y)
 $(info -- Supported Unofficial Manager: MamboSU, KowSU, MKSU, RKSU, SukiSU-Ultra, ReSukiSU)
diff --git a/kernel/manager_sign.h b/kernel/manager_sign.h
index a0db171..92478a1 100644
--- a/kernel/manager_sign.h
+++ b/kernel/manager_sign.h
@@ -23,6 +23,11 @@
 #define EXPECTED_HASH_KOWX712                                                  \
     "484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588"
 
+// RapliVx/KernelSU (MamboSU)
+#define EXPECTED_SIZE_MAMBO 0x384
+#define EXPECTED_HASH_MAMBO \
+	"a9462b8b98ea1ca7901b0cbdcebfaa35f0aa95e51b01d66e6b6d2c81b97746d8"
+
 typedef struct {
     u32 size;
     const char *sha256;
diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index 6890889..661c56c 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -29,6 +29,7 @@ static apk_sign_key_t apk_sign_keys[] = {
     { EXPECTED_SIZE_RSUNTK, EXPECTED_HASH_RSUNTK }, // RKSU
     { EXPECTED_SIZE_5EC1CFF, EXPECTED_HASH_5EC1CFF }, // MKSU
     { EXPECTED_SIZE_KOWX712, EXPECTED_HASH_KOWX712 }, // KowSU
+	{ EXPECTED_SIZE_MAMBO, EXPECTED_HASH_MAMBO }, // MamboSU
 #ifdef EXPECTED_SIZE
     { EXPECTED_SIZE, EXPECTED_HASH }, // Custom
 #endif
