diff --git a/kernel/Kbuild b/kernel/Kbuild
index abd1380..6274d22 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -64,6 +64,13 @@ $(info -- $(REPO_NAME) version name: $(KSU_VERSION_FULL))
 
 ccflags-y += -DKSU_VERSION=$(KSU_VERSION)
 ccflags-y += -DKSU_VERSION_FULL=\"$(KSU_VERSION_FULL)\"
+
+# Define Manager List automatically if not set
+ifndef MamboSU_MANAGER_LIST
+# List: Official, KOWX712, RSUNTK, 5EC1CFF, MamboSU, Next, ShirkNeko, Neko, ReSukiSU
+MamboSU_MANAGER_LIST := 0x033b:c371061b19d8c7d7d6133c6a9bafe198fa944e50c1b31c9d8daa8d7f1fc2d2d6,0x375:484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588,0x396:f415f4ed9435427e1fdf7f1fccd4dbc07b3d6b8751e4dbcec6f19671f427870b,0x180:7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4,0x384:a9462b8b98ea1ca7901b0cbdcebfaa35f0aa95e51b01d66e6b6d2c81b97746d8,0x3e6:79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7,0x35c:947ae944f3de4ed4c21a7e4f7953ecf351bfa2b36239da37a34111ad29993eef,0x29c:946b0557e450a6430a0ba6b6bccee5bc12953ec8735d55e26139b0ec12303b21,0x377:d3469712b6214462764a1d8d3e5cbe1d6819a0b629791b9f4101867821f1df64
+endif
+
+ccflags-y += -DKSU_NEXT_MANAGER_LIST=\"$(MamboSU_MANAGER_LIST)\"
 
 ifeq ($(CONFIG_KSU_MULTI_MANAGER_SUPPORT), y)
-$(info -- Supported Unofficial Manager: MKSU, KowSU, RKSU, SukiSU-Ultra, ReSukiSU )
+$(info -- Supported Unofficial Manager: MKSU, KowSU, RKSU, SukiSU-Ultra, ReSukiSU, MamboSU )
 endif